e2e/tasks/setAllPoolsMinDelegationTimeZero.js

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env node
+
+const { ethers } = require('ethers');
+const { parseError } = require('../parseError');
+const { gasLog } = require('../gasLog');
+
+const log = require('debug')(`e2e:${require('path').basename(__filename, '.js')}`);
+
+async function setAllPoolsMinDelegationTimeZero({ poolId }) {
```

Comment:
Making this smaller "task" will make it more reusable.

```suggestion
async function setMarketMinDelegationTime({ poolId, marketId, delegationTime }) {
```

This seems to be a specific stuff for legacy market migration so fetching markets and iterating over them and calling `setMarketMinDelegationTime` would suit the code of test itself better, than task.

This is more of a suggestion as tasks are meant to be more of a "reusable" smaller blocks of work that don't do more than one thing

Reply:
this task would be difficult to "reuse" outside of testing context because in normal operation it can only be executed by the market contract itself.

and I cant think of a situation where we would not want to set all min delgation times to any value other than 0, especially since we are getting close to phasing this system out.

Reply:
situation to use non-zero value is to set it back to non zero value after setting it to zero value =)